testutil/compose/auto.go

```diff
@@ -252,9 +261,40 @@ func NewRunnerFunc(topic string, dir string, up bool, runFunc RunFunc,
 		}
 
 		if up {
-			return data, execUp(ctx, dir)
+			return data, executeAndLogCompose(ctx, dir, logFile, topic)
 		}
 
 		return data, nil
 	}
 }
+
+// prepLogFile returns provided logFile as writer with step string appended to it.
+func prepLogFile(logFile string, step string) (io.WriteCloser, error) {
+	file, err := os.OpenFile(logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644) //nolint:nosnakecase
+	if err != nil {
+		return file, errors.Wrap(err, "open log file")
+	}
+
+	_, err = file.WriteString(fmt.Sprintf("=== STEP: %s ===\n", step))
+	if err != nil {
+		return file, errors.Wrap(err, "write string to file")
+	}
+
+	return file, nil
+}
+
+// executeAndLogCompose prepares log file (if logFile provided) and executes docker compose up command.
+func executeAndLogCompose(ctx context.Context, dir string, logFile string, step string) error {
```

Comment:
added a separate function to handle file handling (file open and close) so that it isn't repetitive to call `file.Close()` whenever `execUp` is called